test/e2e/main/main_test.go

```diff
@@ -1450,3 +1451,63 @@ func createCacheWithCR(cacheName string, nameSpace string, clusterName string) *
 		},
 	}
 }
+
+func TestPodDegradationAfterOOM(t *testing.T) {
+	t.Parallel()
+
+	//Creating Infinispan cluster
+	ispn := tutils.DefaultSpec(testKube)
+	ispn.Name = strcase.ToKebab(t.Name())
+	ispn.Labels = map[string]string{"test-name": t.Name()}
+	ispn.Spec.Replicas = 2
+	ispn.Spec.Container.Memory = "256Mi"
+	ispn.Spec.Security.EndpointAuthentication = pointer.BoolPtr(false)
+
+	testKube.CreateInfinispan(ispn, tutils.Namespace)
+	defer testKube.CleanNamespaceAndLogOnPanic(tutils.Namespace, ispn.Labels)
+	testKube.WaitForInfinispanPods(int(ispn.Spec.Replicas), tutils.SinglePodTimeout, ispn.Name, tutils.Namespace)
+	testKube.WaitForInfinispanCondition(ispn.Name, ispn.Namespace, ispnv1.ConditionWellFormed)
+
+	//Creating cache
+	cacheName := "failover-cache"
+	template := `<infinispan><cache-container><replicated-cache name ="` + cacheName +
+		`"><encoding media-type="text/plain"/></replicated-cache></cache-container></infinispan>`
+	veryLongValue := GenerateStringWithCharset(6000)
+
+	hostAddr, client := tutils.HTTPClientAndHost(ispn, testKube)
+	createCacheWithXMLTemplate(cacheName, hostAddr, template, client)
+
+	//Generate tons of random entries
+	for key := 1; key < 50000; key++ {
+		keyURL := fmt.Sprintf("%v/%v", cacheURL(cacheName, hostAddr), key)
+		headers := map[string]string{"Content-Type": "text/plain"}
+		_, error := client.Post(keyURL, veryLongValue, headers)
+
+		if error != nil {
+			fmt.Printf("ERROR for key=%d, Description=%s", key, error)
+			break
+		}
+	}
+
+	//Check if all pods are running and they are not degraded
+	testKube.WaitForInfinispanPods(int(ispn.Spec.Replicas), tutils.SinglePodTimeout, ispn.Name, tutils.Namespace)
```

Comment:
It would be good if we first asserted that the pods had indeed failed over. Assuming that the very large keys somehow didn't cause the OOM, the test would still pass with these checks.

Reply:
In the updated version I've added an if statement to check whether the pod restarts for an OOM or not.